test/router-luatest/router_test.lua

```diff
@@ -545,3 +545,41 @@ g.test_enable_disable = function(g)
     -- we don't want this server to interfere with subsequent tests
```

Comment:
Apparently CI is failing on the new test. I see the run on the latest master is red.

Reply:
Yep, everything was done with the intention of merging this PR after the appearance of the system fibers in tarantool. Curently if we wanted to keep pursuing the version-specific checks we would have to introduce three (or two but with unnecessary checks) versions of function (just checking for `error` and `closed` - before 2.10.0 and after system fibers, checking error message - 2.10.0, full function you proposed - for 2.10.1).

Let's not do that and just say, that reconnect doesn't happen in 2.10.0. The user can try killing net.box's fibers as much as he wants in the versions, which will be released after introducing system fibers. 

It's not really a bug, and if you think that we should not complicate the code and increase execution time for such insignificant thing, just close this PR without merging.  

But I would really want to decrease `wait_timeout`, it's not very convenient to change it manually every time it's needed. We can make it in another PR, if you think this change is justified.